cloudify_awssdk/common/decorators.py

```diff
@@ -237,3 +259,67 @@ def wrapper_inner(**kwargs):
                 % (resource_type, iface.resource_id, status))
         return wrapper_inner
     return wrapper_outer
+
+
+def check_swift_resource(func):
+    def wrapper(**kwargs):
+        ctx = kwargs['ctx']
+        node_type = ctx.node.type
+        if node_type and node_type.startswith(SWIFT_NODE_PREFIX):
+            response = None
+            swift_config = ctx.node.properties.get('swift_config')
+
+            username = swift_config.get('swift_username')
+            password = swift_config.get('swift_password')
+            auth_url = swift_config.get('swift_auth_url')
+            region_name = swift_config.get('swift_region_name')
+
+            aws_config = {}
+            # Only Generate the token if it is not generated before
+            if not ctx.instance.runtime_properties.get('aws_config'):
+                endpoint_url, token = \
+                    utils.generate_swift_access_config(auth_url,
+                                                       username,
+                                                       password)
+
+                aws_config['aws_access_key_id'] = username
+                aws_config['aws_secret_access_key'] = token
+                aws_config['region_name'] = region_name
+                aws_config['endpoint_url'] = endpoint_url
+                ctx.instance.runtime_properties['aws_config'] = aws_config
+
+            try:
+                kwargs['aws_config'] = aws_config
+                kwargs['ctx'] = ctx
+                response = func(**kwargs)
+            except ClientError as err:
```

Comment:
There is no proper reporting of the error.
We raise `OperationRetry` below, which is good; but the original error is lost forever.
I'm not sure if `OperationRetry` supports the `causes` attribute.
If it does, then please use it.
Otherwise, please log the error to the context logger.

Reply:
`OperationRetry` supports `causes`, I  updated that as well